sys/sys/mman.h

```diff
@@ -56,8 +56,26 @@
 #define	PROT_WRITE	0x02	/* pages can be written */
 #define	PROT_EXEC	0x04	/* pages can be executed */
 #if __BSD_VISIBLE
-#define	_PROT_ALL	(PROT_READ | PROT_WRITE | PROT_EXEC)
-#define	PROT_EXTRACT(prot)	((prot) & _PROT_ALL)
+/*
+ * For backwards capability load and store protections are implied by the
+ * values of PROT_READ and PROT_WRITE or-ed together unless any PROT_CAP_*
+ * value is included.
+ */
+#define	PROT_CAP_NONE	0x8000
```

Comment:
This value is awkward for PROT_MAX given the underlying type is `int` not `unsigned int`, since it ends up being shifted into the sign bit which is UB.

Reply:
Yeah, there are a few options here. It could be `0x20`, but that overlaps with an assigned (in-kernel) VM_PROT value so there's some risk of errors in future interface implementations. Maybe if should be 0x4000 to avoid the UB?